hail_search/hail_search_query.py

```diff
@@ -642,18 +754,29 @@ class VariantHailTableQuery(BaseHailTableQuery):
 
     @staticmethod
     def _selected_main_transcript_expr(ht):
-        has_allowed_transcripts = 'allowed_transcripts' in ht.row
+        gene_id = getattr(ht, 'gene_id', None)
+        if gene_id is not None:
+            gene_transcripts = ht.sorted_transcript_consequences.filter(lambda t: t.gene_id == ht.gene_id)
+        else:
+            gene_transcripts = getattr(ht, 'gene_transcripts', None)
+
+        allowed_transcripts = getattr(ht, 'allowed_transcripts', None)
+        if hasattr(ht, 'has_allowed_annotation_secondary'):
+            allowed_transcripts = hl.if_else(
+                allowed_transcripts.any(hl.is_defined), allowed_transcripts, ht.allowed_transcripts_secondary,
+            ) if allowed_transcripts is not None else ht.allowed_transcripts_secondary
+
         main_transcript = ht.sorted_transcript_consequences.first()
-        if 'gene_transcripts' in ht.row:
-            matched_transcript = ht.gene_transcripts.first()
-            if has_allowed_transcripts:
-                allowed_transcript_ids = hl.set(ht.allowed_transcripts.map(lambda t: t.transcript_id))
+        if gene_transcripts is not None:
```

Comment:
This might be cleaner as 

`if gene_transcripts is not None and allowed_transcripts is not None:`

to save an indentation block.